controller/environment.go

```diff
@@ -0,0 +1,154 @@
+		Controller:           service.NewController("EnvironmentController"),
+		db:                   db,
+		developerModeEnabled: developerModeEnabled,
+	}
+}
+
+func ConvertEnvironment(env *environment.Environment) *app.Environment {
+	respEnv := &app.Environment{
+		ID:   env.ID,
+		Type: APIStringTypeEnvironment,
+		Attributes: &app.EnvironmentAttributes{
+			Name:          env.Name,
+			Type:          env.Type,
+			NamespaceName: env.NamespaceName,
+			ClusterURL:    env.ClusterURL,
+		},
+		// TODO add links, relations
+	}
+	return respEnv
+}
+
+func ConvertEnvironments(envs []*environment.Environment) *app.EnvironmentsList {
+	res := &app.EnvironmentsList{Data: make([]*app.Environment, len(envs), len(envs))}
+	for ind, env := range envs {
+		res.Data[ind] = ConvertEnvironment(env)
+	}
+	return res
+}
+
+func (c *EnvironmentController) Create(ctx *app.CreateEnvironmentContext) error {
+	if !c.developerModeEnabled {
+		log.Error(ctx, nil, "operation not allowed")
+		err, _ := app.ErrorToJSONAPIErrors(ctx, errors.NewInternalErrorFromString("operation not allowed"))
+		return ctx.MethodNotAllowed(err)
+	}
+
+	tokenMgr, err := token.ReadManagerFromContext(ctx)
+	if err != nil {
+		return app.JSONErrorResponse(ctx, err)
+	}
+	_, err = tokenMgr.Locate(ctx)
+	if err != nil {
+		return app.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
+
+	reqEnv := ctx.Payload.Data
+	if reqEnv == nil {
+		return app.JSONErrorResponse(ctx, errors.NewBadParameterError("data", nil).Expected("not nil"))
+	}
+	spaceID := ctx.SpaceID
+	err = c.checkSpaceExist(ctx, spaceID.String())
+	if err != nil {
+		return app.JSONErrorResponse(ctx, err)
+	}
+
+	var env *environment.Environment
+	err = application.Transactional(c.db, func(appl application.Application) error {
+		newEnv := environment.Environment{
+			Name:          reqEnv.Attributes.Name,
+			Type:          reqEnv.Attributes.Type,
+			SpaceID:       &spaceID,
+			NamespaceName: reqEnv.Attributes.NamespaceName,
+			ClusterURL:    reqEnv.Attributes.ClusterURL,
+		}
+
+		env, err = appl.Environments().Create(ctx, &newEnv)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{"err": err},
+				"failed to create environment: %s", newEnv.Name)
+			return errs.Wrapf(err, "failed to create environment: %s", newEnv.Name)
+		}
+		return nil
+	})
+	if err != nil {
+		return app.JSONErrorResponse(ctx, err)
+	}
+
+	envData := ConvertEnvironment(env)
+	res := &app.EnvironmentSingle{
+		Data: envData,
+	}
+	ctx.ResponseData.Header().Set("Location", httpsupport.AbsoluteURL(&goa.RequestData{Request: ctx.Request},
+		app.EnvironmentHref(res.Data.ID), nil))
+	return ctx.Created(res)
+}
+
+func (c *EnvironmentController) List(ctx *app.ListEnvironmentContext) error {
+	tokenMgr, err := token.ReadManagerFromContext(ctx)
+	if err != nil {
+		return app.JSONErrorResponse(ctx, err)
+	}
+	_, err = tokenMgr.Locate(ctx)
+	if err != nil {
+		return app.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
```

Comment:
Why you load token manager here and above? Just for future use? JWT middleware will verify token signature and expiration for you.
But we will need to check if the user has permissions to list the environments (space collaborator). And we will need the token for that. So, just curious is this token manager thing here just for future use?